internal/pkg/agent/application/coordinator/state/state.go

```diff
@@ -0,0 +1,365 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package state
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	agentclient "github.com/elastic/elastic-agent/pkg/control/v2/client"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent-libs/logp"
+	"github.com/elastic/elastic-agent/pkg/component/runtime"
+)
+
+// State provides the current state of the coordinator along with all the current states of components and units.
+type State struct {
+	State        agentclient.State                 `yaml:"state"`
+	Message      string                            `yaml:"message"`
+	FleetState   agentclient.State                 `yaml:"fleet_state"`
+	FleetMessage string                            `yaml:"fleet_message"`
+	Components   []runtime.ComponentComponentState `yaml:"components"`
+	LogLevel     logp.Level                        `yaml:"log_level"`
+}
+
+// StateFetcher provides an interface to fetch the current state of the coordinator.
+type StateFetcher interface {
```

Comment:
I realize that this interface was already in the coordinator package before this change, but why is this interface here?
If it's something defined by the usage in the FleetGateway, shouldn't this live in that package ?